### ISSUE TITLE
Add grain and vignette to the env post-processing. (and more)

### ISSUE THREAD

**Issue by y0nd0** [godotengine/godot-proposals#1684]
I like the env post-processing in Godot 3.2.3. It's easy to enable without additional plugins. In Unity there are some more effects like "Grain" and "Vignette". Also "Motion Blur". This issue is about to extend the official build-in post-processing. Effects to make the picture more "realistic". I mean, this post-effects can improve the impression of it. 

For the implementation, we could play with these effects in Unity. I don't want to compare Unity with Godot. It's just another engine which has these effects. Just an example.
Short description:

| Effect | Description |
| --- | --- |
| **Grain** | Film grain is the random optical texture of photographic film. For example a perlin noise. The size in this picture is maybe too big, but a good demonstration. We need control of the size, opacity, movement etc. ![grain](https://user-images.githubusercontent.com/68586847/96365853-e953db00-1143-11eb-86d0-74c4c1dd729e.jpg) ![demo](https://user-images.githubusercontent.com/68586847/96366669-528a1d00-1149-11eb-978b-e3929a16c323.gif) (Note: Gif quality sucks.) | 
| **Vignette** | In Photography, vignetting is the term used for the darkening and/or desaturating towards the edges of an image compared to the center. We need control the size, smoothness and opacity. ![vignette](https://user-images.githubusercontent.com/68586847/96365922-75660280-1144-11eb-80b3-60660769c837.jpg) |
| **Motion Blur** | Motion Blur is a common post-processing effect that simulates the blurring of an image when objects filmed by a camera are moving faster than the camera’s exposure time. (Optional for this issue, but would be great.)<br/> ![motion-blur](https://user-images.githubusercontent.com/68586847/96365986-eefdf080-1144-11eb-9392-b4e25c934de6.gif) |
| **Lens Dirt** | Lens Dirt applies a fullscreen layer of smudges or dust to diffract the Bloom effect. This is commonly used in modern first person shooters. Also optional for this issue. But another cool thing. ![image](https://user-images.githubusercontent.com/68586847/96366142-fb367d80-1145-11eb-8a40-e7bff896297a.png) This is set in the bloom / glow section,  A texture like: ![image](https://user-images.githubusercontent.com/68586847/96366180-389b0b00-1146-11eb-83b6-5010c83e69e2.png) ![demo](https://user-images.githubusercontent.com/68586847/96366357-89f7ca00-1147-11eb-9d5a-1613f021ed1b.gif) (The dirt lens, not the lens flares.) |

**Comment by Zireael07** [comment]
There's an asset that does lens flare effects (without the dirt)
A vignette effect is showcased in one of the shader effect examples.
As for motion blur, there's an open issue asking for that as well as an asset.

**Comment by y0nd0** [comment]
This issue is about expanding the official post-processing stack. A proposal to integrate it in Godot as build-in effects.

**Comment by Zireael07** [comment]
Since assets can already do most of those, I wouldn't hold my breath for that happening - Godot's philosophy is to keep the core lean, and those are only useful for certain kinds of 3D games and can be implemented fairly easily (except motion blur, which already has an issue open)

**Comment by Calinou** [comment]
In the interest of making high-quality proposals, please [read the rules](https://github.com/godotengine/godot-proposals#readme) and fill in the issue template. This proposal will be reopened once this is done :slightly_smiling_face:

**Comment by SIsilicon** [comment]
> @Zireael07:
> There's an asset that does lens flare effects (without the dirt)

The lens flare vfx plugin _does_ include lens dirt. :(

**Comment by y0nd0** [comment]
Btw. lens flares has nothing to do with lens dirt. It's a bloom feature.

**Comment by y0nd0** [comment]
@Zireael07 

> except motion blur, which already has an issue open

Could you please link this issue? I cannot find it. I know there is an [asset](https://godotengine.org/asset-library/asset/edit/678) for motion blur. But this is no working great.

**Comment by Calinou** [comment]
@y0nd0 There was an [issue on the main repository](https://github.com/godotengine/godot/issues/16301) about motion blur, but it'd need to be reopened on the godot-proposals repository (while following the template).

**Comment by Calinou** [comment]
Superseded by #1806.
